Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using NStack;
+using Terminal.Gui.Trees;
+using static System.Environment;
+using static Terminal.Gui.OpenDialog;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// Modal dialog for selecting files/directories.  Has auto-complete and expandable
+	/// navigation pane (Recent, Root drives etc).
+	/// </summary>
+	public class FileDialog2 : Dialog {
+
+		// TODO : expose these somehow for localization without compromising case/switch statements
+		private const string HeaderFilename = "Filename";
```

Comment:
What are your feelings on using `_` to prefix private/globl vars?

I find it makes code more readable and avoids errors caused by confusing local vars with globals. It's not part of the coding guidelines for this project, but we are not consistent.

Reply:
I love this notation for private fields and have always thought it was clearer and helped avoid mistakes.  I don't know why it is not the standard.  Am happy to change to using it.